client/src/components/Block.tsx

```diff
@@ -2,37 +2,38 @@ import React, { useContext } from "react";
 import "../styling/Maze.css";
 import { PageContext } from "../PageProvider";
 import { Maze, Point } from "../utilities/types";
-import { isValidBoardPoint, isValidEndPoint, isValidStartPoint, resetStartOrEndPoint, switchBlockType, switchStartOrEndPoint, updateMaze } from "../utilities/utilities";
+import { isValidBoardPoint, isValidEndPoint, isValidStartPoint, resetStartOrEndPoint, switchBlockType } from "../utilities/utilities";
 import { MazeContext } from "../MazeProvider";
 
 type BlockProps = {
   blockType: number;
-  maze: Maze;
   rowIndex: number;
   colIndex: number;
 };
 
-const Block: React.FC<BlockProps> = ({ blockType, maze, rowIndex, colIndex }) => {
+const Block: React.FC<BlockProps> = ({ blockType, rowIndex, colIndex }) => {
   const {currentPage} = useContext(PageContext)
-  const {mazeSize, setMaze} = useContext(MazeContext)
+  const {mazeSize, setMaze, pathColor, wallColor, walkedColor, queuedColor, shortPathColor} = useContext(MazeContext)
```

Comment:
Move all the color related states and funcs into a "ThemeProvider". 

Also collapse the many color variables and setState funcs into an object to avoid having to export so many variables from the context